satellite/satellitedb/containment.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"context"
+	"database/sql"
+
+	"storj.io/storj/pkg/audit"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type containment struct {
+	db *dbx.DB
+}
+
+// Get gets the pending audit by node id
+func (containment *containment) Get(ctx context.Context, id pb.NodeID) (*audit.PendingAudit, error) {
+	if id.IsZero() {
+		return nil, audit.ContainError.New("node ID empty")
+	}
+
+	pending, err := containment.db.Get_PendingAudits_By_NodeId(ctx, dbx.PendingAudits_NodeId(id.Bytes()))
+	if err == sql.ErrNoRows {
+		return nil, audit.ErrContainedNotFound.New(id.String())
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return convertDBPending(pending)
+}
+
+// IncrementPending will create a new pending audit entry, or increase its reverify count if it already exists
+func (containment *containment) IncrementPending(ctx context.Context, pendingAudit *audit.PendingAudit) error {
+	statement := containment.db.Rebind(
+		`INSERT INTO pending_audits (node_id, piece_id, stripe_index, share_size, expected_share_hash, reverify_count)
+		VALUES (?, ?, ?, ?, ?, ?)
+		ON CONFLICT(node_id)
+		DO UPDATE SET reverify_count = pending_audits.reverify_count + 1`,
+	)
+	_, err := containment.db.ExecContext(ctx, statement,
+		pendingAudit.NodeID.Bytes(), pendingAudit.PieceID.Bytes(), pendingAudit.StripeIndex, pendingAudit.ShareSize, pendingAudit.ExpectedShareHash, pendingAudit.ReverifyCount,
+	)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Delete deletes the pending audit
+func (containment *containment) Delete(ctx context.Context, id pb.NodeID) error {
+	if id.IsZero() {
+		return audit.ContainError.New("node ID empty")
+	}
+	isDeleted, err := containment.db.Delete_PendingAudits_By_NodeId(ctx, dbx.PendingAudits_NodeId(id.Bytes()))
+	if err == sql.ErrNoRows {
+		return audit.ErrContainedNotFound.New(id.String())
+	}
+	if !isDeleted {
+		return audit.ErrContainDelete.New(id.String(), err)
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func convertDBPending(info *dbx.PendingAudits) (*audit.PendingAudit, error) {
+	if info == nil {
+		return nil, Error.New("missing info")
+	}
+
+	nodeID, err := storj.NodeIDFromBytes(info.NodeId)
+	if err != nil {
+		return nil, err
+	}
+
+	pieceID, err := storj.PieceIDFromBytes(info.PieceId)
+	if err != nil {
+		return nil, err
+	}
+
+	pending := &audit.PendingAudit{
+		NodeID:            nodeID,
+		PieceID:           pieceID,
+		StripeIndex:       int(info.StripeIndex),
```

Comment:
After seen this, I'm wondering if it's a convention to use types without a specific length like `int` rather than the ones with specific length like `int64`

Reply:
I'm not sure if it's a convention. If you think it's better, I can change the StripeIndex type to int32 to be more specific.

Reply:
Well, I always prefer to have explicit length for the variables rather than relying in something that change depending of the architecture, but that's a personal choice.

I never searched before about this topic, but I've just done it now and I found  https://yourbasic.org/golang/int-vs-int64/ that's my same personal opinion.

I cannot ask you to change that, somebody else more familiar with the code base and its evolution since the beginning could do it.

Sometimes, I drop this kind of questions in some reviews when the PR belongs to a person that is working in Storj much earlier than me to figure out things which I'm in a doubt and to learn new things from the answer.

Reply:
I think you're right it's better to be specific, and we know these will be smaller numbers so I went ahead and changed them to type int32.

Reply:
it's an index, so I wouldn't expect negative index, so an unsigned type is better (e.g. `uint32`, `uint16`, etc.)